xDrip/Scenes/Settings/AlertSettings/SettingsAlertSingleType/SettingsAlertSingleTypePresenter.swift

```diff
@@ -22,7 +23,201 @@ final class SettingsAlertSingleTypePresenter: SettingsAlertSingleTypePresentatio
+        }
+        
+        cells.append(
+            createRightSwitchCell(.entireDay, isSwitchOn: config.isEntireDay, switchValueChangedHandler: response.switchValueChangedHandler)
+        )
+        
+        if !config.isEntireDay {
+            let startTimeDate = Date(timeIntervalSince1970: config.startTime)
+            let endTimeDate = Date(timeIntervalSince1970: config.endTime)
+            let startTimeString = DateFormatter.localizedString(from: startTimeDate, dateStyle: .none, timeStyle: .short)
+            let endTimeString = DateFormatter.localizedString(from: endTimeDate, dateStyle: .none, timeStyle: .short)
+            
+            cells.append(contentsOf: [
+                createTimePickerView(.startTime, date: startTimeDate, detailText: startTimeString, valueChangeHandler: response.timePickerValueChangedHandler),
+                createTimePickerView(.endTime, date: endTimeDate, detailText: endTimeString, valueChangeHandler: response.timePickerValueChangedHandler)
+            ])
+        }
+        
+        let highTresholdString = User.current.settings.unit == .mmolL ? "\(config.highThreshold)" : "\(Int(config.highThreshold))"
+        let lowTresholdString = User.current.settings.unit == .mmolL ? "\(config.lowThreshold)" : "\(Int(config.lowThreshold))"
+        cells.append(contentsOf: [
+            createUnitsPickerView(.highTreshold, detailText: highTresholdString, valueChangeHandler: response.pickerViewValueChangedHandler),
+            createUnitsPickerView(.lowTreshold, detailText: lowTresholdString, valueChangeHandler: response.pickerViewValueChangedHandler)
+        ])
+        
+        return .normal(cells: cells, header: "settings_alert_single_type_events_header".localized, footer: nil)
+    }
+    
+    private func createTextInputViewCell(
+        _ field: SettingsAlertSingleType.Field,
+        detailText: String?,
+        placeholder: String?,
+        editingChangedHandler: @escaping (String) -> Void) -> BaseSettings.Cell {
+        
+        return .textInput(mainText: field.title, detailText: detailText, placeholder: placeholder) { (string) in
+            guard let string = string else { return }
+            editingChangedHandler(string)
+        }
+    }
+    
+    private func createRightSwitchCell(
+        _ field: SettingsAlertSingleType.Field,
+        isSwitchOn: Bool,
+        switchValueChangedHandler: @escaping (SettingsAlertSingleType.Field, Bool) -> Void) -> BaseSettings.Cell {
+        
+        return .rightSwitch(text: field.title, isSwitchOn: isSwitchOn) { (value) in
+            switchValueChangedHandler(field, value)
+        }
+    }
+    
+    private func createCountDownPickerView(
+        _ field: SettingsAlertSingleType.Field,
+        detailText: String?,
+        valueChangeHandler: @escaping (SettingsAlertSingleType.Field, TimeInterval) -> Void) -> BaseSettings.Cell {
+        
+        let picker = CustomDatePicker()
+        
+        DispatchQueue.main.async {
```

Comment:
Seems like a dirty hack. If there's really no other way of doing it, move the hack inside `CustomDatePicker` to incapsulate it there and put a very good comment of why it is done that way.